rfcs/0182-user-management.md

```diff
@@ -0,0 +1,350 @@
+  than 7% of the sum of both `nixos + pkgs`. A quick and dirty calculation would
+  bring a similar 7% increase in size.
+
+  This is certainly a crude calculation, however we are ignoring many factors
+  that will bring these numbers down. Namely:
+
+  - Refactorings
+
+    Similar code can be factored and abstracted.
+
+  - Code sharing
+
+    Code that was initially projected for basic user management can be found
+    useful for superuser management too; and vice-versa.
+
+- Code duplications.
+
+  Code duplications can be refactored as they appear.
+
+  Indeed it is way easier to deal with code duplication in a monorepo: since the
+  barrier of communication between multiple repositories disappears, a
+  duplicated code can be factored more easily, requiring just one pull request
+  instead of one per repo.
+
+- Evaluation times of monorepo will increase.
+
+  The increase in evaluation time is a bit harder to measure. However,
+
+  - "increase in evaluation time" was not an argument strong enough for undoing
+    the NixOS assimilation;
+  - arguably, the increase in evaluation time will be felt by those effectively
+    using the hometool; the remaining users will not be signficatively affected.
+  - a similar argument for the increase in size can be crafted here.
+
+  The perceived advantages of having this module system surpasses these small
+  disadvantages about size and evaluation time.
+
+  Further, outside the scope of Nixpkgs, there are initiatives focused on
+  optimizing the Nix evaluators.
+
+# Alternatives
+[alternatives]: #alternatives
+
+## The trivial "do nothing"
+
+Trivially keeping the status quo.
+
+## Promote `guix-home`
+
+What? Lisp is cool!
+
+## Promote `home-manager`
+
+Now a serious alternative worthy of consideration.
+
+Home Manager has the most obvious and the most powerful advantage:
+
+It exists and is working well. Further, it is battle-tested and encodes a lot of
+knowledge in its 8 years lifespan and more than 3.7k commits.
+
+However, it has some non-negligible disadvantages:
+
+- The current code does not follow the same standards of the current Nixpkgs
+  monorepo
+
+  - many modules rely on stringly `extraConfig` instead of structured
+    `settings`, contra RFC 0042;
+
+- Many instances of technical debt, something expected in such a long-lived
+    project:
+
+    - excessive uses of `with lib;`
+
+    - hacky solutions like the dangerous `mkOutOfStoreSymlink`:
+      https://github.com/nix-community/home-manager/issues/3032
+
+    - under-documented workarounds
+
+    - too much reliance on Bash and its nasty idiosyncrasies
+
+On the other hand, starting from a cleaner slate has the opposite set of
+advantages and disadvantages: the kickstart of a project is too wide and it is
+easier to get lost. On the other hand, there is considerably more freedom to
+prototype and test ideas, older and newer.
+
+# Prior art
+[prior-art]: #prior-art
```

Comment:
I believe https://github.com/NixOS/nixpkgs/pull/9250 really deserves a mention here. FWICT it is the closest thing to what is proposed in this RFC.